### PR TITLE
Cleanup PDF output

### LINKED ISSUE #6777: Add copyright handling to contributing doc
Clarify the rules for updating the copyright lines in files when a developer is adding material  (add a new copyright line for the new contributor).

### PR DESCRIPTION
**Pull request by nashif**
- doc: cleanup for PDF
- doc: move kconfig to indices and table
- doc: print tilte of Kconfig on top

Also fixes #6777

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -126,6 +128,8 @@ licensing, as described in :ref:`Zephyr_Licensing`.
    Indices and Tables
    ******************
```
Comment (gmarull) [inline doc/index.rst:124]: should likely be excluded, too

File: doc/_scripts/gen_kconfig_rest.py [hunk 2]
```diff
@@ -528,9 +528,12 @@ def sym_header_rst(sym):
     #
     # - '.. title::' sets the title of the document (e.g. <title>). This seems
     #   to be poorly documented at the moment.
+    l = len(f"CONFIG_{sym.name}")
     return ":orphan:\n\n" \
-           f".. title:: {sym.name}\n\n" \
+           f".. title:: CONFIG_{sym.name}\n\n" \
            f".. kconfig:: CONFIG_{sym.name}\n\n" \
+           f"CONFIG_{sym.name}\n" \
+           f"{'#'*l}\n\n" \
```
Comment (gmarull) [inline doc/_scripts/gen_kconfig_rest.py:536]: nit: `'#' * l`

File: doc/index.rst (hunk 1 above)
Comment (nashif) [reply doc/index.rst]: now we have an index for tex and removing lots of :only: usage in main index

File: doc/index-tex.rst [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+   introduction/index.rst
+   getting_started/index.rst
+   contribute/index.rst
+   development_process/index.rst
+   guides/build/index.rst
+   application/index.rst
+   reference/index.rst
+   guides/index.rst
+   security/index.rst
```
Comment (gmarull) [inline doc/index-tex.rst:24]: I'm not a fan of having two index files, we'll need to maintain 2 duplicate toctrees this way. I also don't see why we can't have the Indices and Tables chapter in the latex output.

Comment (nashif) [reply doc/index-tex.rst:24]: The content is minimal, it is no way the same we maintain in the html variant. If you find a better way to have role based TOC, then I am all ears.

The indices are created automatically, no need to add them. In the html index we are just adding a link to the reference.

**Review by gmarull — APPROVED** [review]

**Review by cfriedt — APPROVED** [review]
LGTM

**Pull request merged**